### PR TITLE
'signdata' RPC call

### PR DESCRIPTION
**Pull request by gandrewstone**
implement an RPC called 'signdata' to generate signatures compatible with the CHECKDATASIG opcode

### CODE REVIEW

**Comment by sickpig** [comment]
why `checkdatasig-activation.py` has been removed?

**Comment by gandrewstone** [comment]
It hasn't.  Its been renamed with an underscore rather than a dash, and I added signdata tests to it.

**Comment by gandrewstone** [comment]
accident, thanks!  I renamed it so that python import works but did not git add the file.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -558,6 +559,104 @@ UniValue signmessage(const UniValue &params, bool fHelp)
...
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 3 || params.size() > 4)
+        throw runtime_error(
+            "signdata \"bitcoinaddress\" \"msgFormat\" \"message\"\n"
+            "\nSign message for use with the CHECKDATASIG instruction."
+            "\nAs per the CHECKDATASIG operation, this RPC normally signs the SHA256 of the provided message\n"
+            "\nunless the 'hash' message format is specified.\n" +
+            HelpRequiringPassphrase() +
```
Comment (ptschip) [inline src/wallet/rpcwallet.cpp:575]: nit: seems you've got a double \n here...

-\nunless the 'hash' message format....
+unless the'hash' message format...

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -558,6 +559,104 @@ UniValue signmessage(const UniValue &params, bool fHelp)
...
+            "\n"
+            "\nArguments:\n"
+            "1. \"bitcoinaddress\"  (string, required) The bitcoin address to use for the private key.\n"
+            "2. \"msgFormat\"       (string, required) Use \"string\", \"hex\", or \"hash\" to specify the message "
+            "encoding.\n"
```
Comment (ptschip) [inline src/wallet/rpcwallet.cpp:580]: I wasn't clear that this "hash" was a 256bit hash only and tried shorter hashes which were rejected.  Maybe a comment there to clarify it to the user?

**Comment by ptschip** [comment]
Other than my two comments, it looks good.

**Review by ptschip — APPROVED** [review]

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (gandrewstone) [reply src/wallet/rpcwallet.cpp:580]: fixed

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (gandrewstone) [reply src/wallet/rpcwallet.cpp:575]: fixed

**Pull request merged**